src/components/ItemList/index.jsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState, useEffect } from 'react';
```

Comment:
I feel like there is room for improvement in this file. 
The idea is that there are only 2 behaviors that are going to trigger a change in the checkbox: 
1) The database is going to tell us that the item was purchased in the last 24 hours (And I agree with @alxmcr suggestion, this needs to be down to the minute / millisecond, because conventionally milliseconds are used for comparison, so instead of the function of comparing hours, go with millisecond based comparison) 
2) The user clicks the checkbox, we mark it checked and disable it. 

For (1) I feel like you are doing too much, since you are getting the lastPurchasedDate as a property, so you could calculate the initialState in line 6, and set it as an initialState. You could use a function for this as well, it could be in a Utils/date.js file isWithin24Hours(lastPurchasedDate)

For (2) HandleCheckbox does a good job of this. But I will do some renaming of some functions 
EditProduct -> MarkProductPurchased (Editing the product sounds more of like I made a typo on the name of the product, or you are editing every aspect of a product, in this case you are only updating the lastPurchasedDate)

And you have a useEffect, that depends on changes to the isChecked, docId, and token. I am not sure I see when this will be called or why, you already called the database with the lastUpdatedDate in the handleCheckbox, you marked the product as purchased and disable it. When is this useEffect being used ?

Reply:
Hello, @indiragp, changes were already implemented.